rebase PR #422

JohnRDOrazio opened the pull request:
Replaces PR #422 linked to https://core.trac.wordpress.org/ticket/50493 .

JohnRDOrazio committed:
rebase PR #422

TimothyBJacobs commented on src/wp-includes/rest-api.php:
```diff
@@ -38,26 +38,55 @@ function register_rest_route( $route_namespace, $route, $args = array(), $overri
-				__( 'REST API routes must be registered on the %s action.' ),
-				'<code>rest_api_init</code>'
+				/* translators: 1: rest_api_init, 2: string value of the namespace, 3: string value of the route. */
+				__( 'REST API routes must be registered on the %1$s action. Instead route \'%3$s\' with namespace \'%2$s\' was not registered on this action.' ),
```
This still needs to have the order of placeholders match their appearance order.

JohnRDOrazio replied:
done. Order of strings in a single translation has been set to be preferable to the order of the parameters in all other surrounding translations.

JohnRDOrazio committed:
swap string values
it's preferable that string values in sprintf statements be in the same order as the translation string rather than follow the same order of all the other strings in the same class / file

JohnRDOrazio committed:
swap variables order too!

JohnRDOrazio commented:
@TimothyBJacobs fixed

TimothyBJacobs commented:
Merged in [r58933](https://core.trac.wordpress.org/changeset/58933).

Thanks for the PR!
